src/vs/workbench/services/search/node/fileSearch.ts

```diff
@@ -659,12 +659,19 @@ export class Engine implements ISearchEngine<IRawFileMatch> {
 		this.walker = new FileWalker(config);
 	}
 
-	public search(onResult: (result: IRawFileMatch) => void, onProgress: (progress: IProgress) => void, done: (error: Error, complete: ISerializedSearchComplete) => void): void {
-		this.walker.walk(this.folderQueries, this.extraFiles, onResult, (err: Error, isLimitHit: boolean) => {
-			done(err, {
-				limitHit: isLimitHit,
-				stats: this.walker.getStats()
-			});
+	search(): PPromise<ISerializedSearchComplete, ISearchProgress<IRawFileMatch>> {
+		return new PPromise<ISerializedSearchComplete, ISearchProgress<IRawFileMatch>>((complete, error, progress) => {
+			this.walker.walk(this.folderQueries, this.extraFiles,
```

Comment:
Oh! yeah, I forgot to mention this in the PR description. This usage was very odd. Will try migrating this too.